src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
```

Comment:
**Issue**: The method setCapacity(int newCapacity) modifies the capacity of the cache without properly handling the eviction of entries when the new capacity is less than the current size of the cache. This can lead to a situation where the cache size exceeds the specified capacity, which is a critical issue for the functioning of an LRU cache. <br> **Fix**: Implement eviction logic within the setCapacity method to ensure that the cache does not exceed the new capacity. This involves evicting the least recently used items until the cache size is equal to or less than the new capacity. <br> **Code Suggestion**: 
 ```
 @@ -22,5 +22,10 @@
 private void setCapacity(int newCapacity) {
     checkCapacity(newCapacity);
 +    while (data.size() > newCapacity) {
 +        Entry<K, V> evicted = evict();
 +        data.remove(evicted.getKey());
 +    }
     this.cap = newCapacity;
 }
 ```